### PR TITLE
Fixes #19521 - Logrotate for smart_proxy_dynflow_core

### PR DESCRIPTION
**Pull request by adamruzicka**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [x] 1.15
* [ ] 1.14
* [ ] 1.13

See Foreman's [plugin maintainer documentation](http://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

File: rubygem-smart_proxy_dynflow_core/logrotate.conf [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/var/log/foreman-proxy/smart_proxy_dynflow_core.log {
+  missingok
+  notifempty
+  create 0644 foreman-proxy foreman-proxy
+  sharedscripts
+  rotate 5
+  compress
+        daily
```
Comment (domcleal) [inline rubygem-smart_proxy_dynflow_core/logrotate.conf]: Inconsistent indentation

File: rubygem-smart_proxy_dynflow_core/rubygem-smart_proxy_dynflow_core.spec [hunk 2]
```diff
@@ -15,11 +15,12 @@
 Summary: Core Smart Proxy Dynflow Service
 Name: %{?scl_prefix}rubygem-%{gem_name}
 Version: 0.1.6
-Release: 1%{?foremandist}%{?dist}
+Release: 2%{?foremandist}%{?dist}
```
Comment (domcleal) [inline rubygem-smart_proxy_dynflow_core/rubygem-smart_proxy_dynflow_core.spec]: Don't modify this.

File: rubygem-smart_proxy_dynflow_core/rubygem-smart_proxy_dynflow_core.spec [hunk 3]
```diff
@@ -109,6 +111,7 @@ install -Dp -m0644 %{buildroot}%{gem_instdir}/deploy/%{service_name}.service %{b
 %{root_sysconfdir}/%{gem_name}/settings.yml
 %doc %{gem_instdir}/LICENSE
 %{root_bindir}/%{service_name}
+%config(noreplace) %{_sysconfdir}/logrotate.d/%{name}
```
Comment (domcleal) [inline rubygem-smart_proxy_dynflow_core/rubygem-smart_proxy_dynflow_core.spec]: Why noreplace?

Comment (adamruzicka) [inline rubygem-smart_proxy_dynflow_core/rubygem-smart_proxy_dynflow_core.spec]: Smart-proxy does it this way and it felt right to do treat it the same way here.

Comment (domcleal) [inline rubygem-smart_proxy_dynflow_core/rubygem-smart_proxy_dynflow_core.spec]: I don't think I agree, this file should be updated if possible, I don't think it makes sense for a user to manage it.

Comment (adamruzicka) [inline rubygem-smart_proxy_dynflow_core/rubygem-smart_proxy_dynflow_core.spec]: Right, removed it

File: rubygem-smart_proxy_dynflow_core/rubygem-smart_proxy_dynflow_core.spec [hunk 4]
```diff
@@ -83,8 +84,9 @@ cp -pa .%{_bindir}/* %{buildroot}%{root_bindir}/
 mkdir -p %{buildroot}%{root_sysconfdir}/smart_proxy_dynflow_core
 cp -pa .%{gem_instdir}/config/settings.yml.example %{buildroot}%{root_sysconfdir}/smart_proxy_dynflow_core/settings.yml
 
-#copy init scripts and sysconfigs
+#copy init scripts, sysconfigs and logrotate config
 install -Dp -m0644 %{buildroot}%{gem_instdir}/deploy/%{service_name}.service %{buildroot}%{_unitdir}/%{service_name}.service
+install -Dp -m0644 %{SOURCE1} %{buildroot}%{_sysconfdir}/logrotate.d/%{name}
```
Comment (domcleal) [inline rubygem-smart_proxy_dynflow_core/rubygem-smart_proxy_dynflow_core.spec]: Is `_sysconfdir` correct in an SCL environment? Wouldn't this be within the SCL root? Is that supported?

Comment (adamruzicka) [inline rubygem-smart_proxy_dynflow_core/rubygem-smart_proxy_dynflow_core.spec]: You're right, fixed

**Comment by domcleal** [comment]
Merged, thanks. Please make the same change to deb/develop for consistency.

**Comment by domcleal** [comment]
Oh, please also squash commits in future - having a commit per review comment is not usual in Foreman repositories.

**Pull request merged**